internal/commands/vaultsecrets/secrets/open.go

```diff
@@ -120,42 +119,55 @@ func openRun(opts *OpenOpts) (err error) {
 		return fmt.Errorf("failed to read the secret %q: %w", opts.SecretName, err)
 	}
 
-	var secretValue string
+	var (
+		secretValue string
+		field       format.Field
+	)
 
 	switch {
 	case resp.Payload.Secret.RotatingVersion != nil:
 		secretValue, err = formatSecretMap(resp.Payload.Secret.RotatingVersion.Values)
 		if err != nil {
 			secretValue = "<< COULD NOT ENCODE TO JSON >>"
 		}
+
+		field = format.Field{
+			Name:        "Values",
+			ValueFormat: `{{ range $key, $value := .RotatingVersion.Values }}{{printf "%s: %s\n" $key $value}}{{ end }}`,
+		}
```

Comment:
if `formatSecretMap` encodes the map to JSON, why do we also need the format field here?

Reply:
The output of `formatSecretMap` is assigned to `secretValue` which is only used when the secret values are written to a file. `secretValue` is not used otherwise.

The format field here defines how we want to template the secret values for each type when its displayed in the terminal.